server/controllers/shelves/delete_by_ids.js

```diff
@@ -21,29 +20,29 @@ module.exports = (req, res, next) => {
   .catch(error_.Handler(req, res))
 }
 
-const deleteByIds = params => {
+const deleteByIds = async params => {
   const { ids, reqUserId, withItems } = params
-  return shelves_.byIdsWithItems(ids)
-  .then(_.compact)
-  .then(tap(validateDeletion(withItems)))
-  .then(tap(validateOwnership(reqUserId)))
-  .then(tap(deleteShelfItems(withItems)))
-  .then(shelves_.bulkDelete)
+  const shelvesRes = await shelves_.byIdsWithItems(ids)
+  const shelves = _.compact(shelvesRes)
+  validateDeletion(withItems, shelves)
+  validateOwnership(reqUserId, shelves)
+  deleteShelfItems(withItems, shelves)
+  shelves_.bulkDelete(shelves)
```

Comment:
```suggestion
  await deleteShelfItems(withItems, shelves)
  return shelves_.bulkDelete(shelves)
```
the ok response would be returned before anything is actually deleted without those